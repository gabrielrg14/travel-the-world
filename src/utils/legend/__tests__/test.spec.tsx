import { legendIcons, legendLinks } from 'utils'

describe('legend', () => {
  describe('legendIcons', () => {
    it('should match snapshot', () => {
      expect(legendIcons).toMatchSnapshot()
    })

    it('should contain home, car, airplane and target icons', () => {
      expect(legendIcons).toContainEqual({
        name: 'home',
        title: 'Home',
        text: 'My place',
        label: 'Maps and location icons created by juicy_fish - Flaticon',
        href: 'https://www.flaticon.com/free-icons/maps-and-location'
      })

      expect(legendIcons).toContainEqual({
        name: 'car',
        title: 'Car',
        text: 'Visited by car',
        label: 'Maps and location icons created by juicy_fish - Flaticon',
        href: 'https://www.flaticon.com/free-icons/maps-and-location'
      })

      expect(legendIcons).toContainEqual({
        name: 'airplane',
        title: 'Airplane',
        text: 'Visited by airplane',
        label: 'Airplane icons created by juicy_fish - Flaticon',
        href: 'https://www.flaticon.com/free-icons/airplane'
      })

      expect(legendIcons).toContainEqual({
        name: 'target',
        title: 'Target',
        text: 'To be visited',
        label: 'Goal icons created by juicy_fish - Flaticon',
        href: 'https://www.flaticon.com/free-icons/goal'
      })
    })
  })

  describe('legendLinks', () => {
    it('should match snapshot', () => {
      expect(legendLinks).toMatchSnapshot()
    })

    it('should contain juicy_fish and flaticon links', () => {
      expect(legendLinks).toContainEqual({
        title: 'juicy_fish icons',
        text: 'juicy_fish',
        label: 'Go to juicy_fish icons',
        href: 'https://www.flaticon.com/authors/juicy-fish'
      })

      expect(legendLinks).toContainEqual({
        title: 'Flaticon',
        text: 'Flaticon',
        label: 'Go to Flaticon',
        href: 'https://www.flaticon.com'
      })
    })
  })
})
