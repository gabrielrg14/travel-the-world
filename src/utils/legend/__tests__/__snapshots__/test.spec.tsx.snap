// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`legend legendIcons should match snapshot 1`] = `
[
  {
    "href": "https://www.flaticon.com/free-icons/maps-and-location",
    "label": "Maps and location icons created by juicy_fish - Flaticon",
    "name": "home",
    "text": "My place",
    "title": "Home",
  },
  {
    "href": "https://www.flaticon.com/free-icons/maps-and-location",
    "label": "Maps and location icons created by juicy_fish - Flaticon",
    "name": "car",
    "text": "Visited by car",
    "title": "Car",
  },
  {
    "href": "https://www.flaticon.com/free-icons/airplane",
    "label": "Airplane icons created by juicy_fish - Flaticon",
    "name": "airplane",
    "text": "Visited by airplane",
    "title": "Airplane",
  },
  {
    "href": "https://www.flaticon.com/free-icons/goal",
    "label": "Goal icons created by juicy_fish - Flaticon",
    "name": "target",
    "text": "To be visited",
    "title": "Target",
  },
]
`;

exports[`legend legendLinks should match snapshot 1`] = `
[
  {
    "href": "https://www.flaticon.com/authors/juicy-fish",
    "label": "Go to juicy_fish icons",
    "text": "juicy_fish",
    "title": "juicy_fish icons",
  },
  {
    "href": "https://www.flaticon.com",
    "label": "Go to Flaticon",
    "text": "Flaticon",
    "title": "Flaticon",
  },
]
`;
